book/07-extending-learners.qmd

```diff
@@ -5,47 +5,53 @@ library(mlr3)
+If you think the learner will be useful to others as well, you can consider creating a pull request in the [mlr3extralearners](https://github.com/mlr-org/mlr3extralearners) repository.
+If you intend to add a learner to mlr3extralearners, **it is strongly recommended** that you **first** open a [learner request issue](https://github.com/mlr-org/mlr3extralearners/issues/new?assignees=&labels=new+learner&template=learner-request-template.md&title=%5BLRNRQ%5D+Add+%3Calgorithm%3E+from+package+%3Cpackage%3E) to discuss the learner you want to implement.
+This allows us to discuss the purpose and necessity of the learner before you start to put the real work in!
+Note that some of the steps shown are only applicable in case you plan on creating a pull request to the mlr-org.
 
 This section gives insights on how a mlr3learner is constructed and how to troubleshoot issues.
 See the [Learner FAQ subsection](#learner-faq) for help.
 
 **Summary of steps for adding a new learner**
 
-1. Check the learner does not already exist [here](https://mlr3extralearners.mlr-org.com/articles/learners/list_learners.html).
-2. [Fork, clone and load](#setup) `r mlr3book::mlr_pkg("mlr3extralearners")`.
+1. Check that the learner does not already exist [here](https://mlr3extralearners.mlr-org.com/articles/learners/list_learners.html).
+2. [Install](#setup) `r mlr3book::mlr_pkg("mlr3extralearners")` and if you want to create a PR, also [fork and clone](#setup) it.
 3. [Run](#create-learner) `r ref("mlr3extralearners::create_learner()")`.
-4. Add the learner [`param_set`](#param-set).
+4. Add the learner's [`param_set`](#param-set).
 5. Manually add [`.train`](#learner-train) and [`.predict`](#learner-predict) private methods to the learner.
-6. If applicable add `importance` and `oob_error` public methods to the learner.
-7. If applicable add references to the learner.
-8. Check [unit tests](#learner-test) and [paramtests](#learner-test) pass (these are automatically created).
-9. Run [cleaning functions](#cleaning)
-10. Open a [pull request](https://github.com/mlr-org/mlr3extralearners/pulls) with the new learner template.
-
-**(Do not copy/paste the code shown in this section. Use `r ref("create_learner()")` to start.)**
+6. If the learner has the property
 
-## Setting-up mlr3extralearners {#setup}
+    *  `"importance"` add the method `importance()`
+    * `"oob_error"` add the method `oob_error()`
+    * `"loglik"` add the method `loglik()`
+    * `"selected_features"` add the method `selected_features()`
+    *  `"hotstart_{forward, backward}"` add the private method `.hotstart()`
 
-In order to use the `r ref("mlr3extralearners::create_learner")` function you must have a local copy of the `r mlr3book::mlr_pkg("mlr3extralearners")` repository and must specify the correct path to the package. To do so, follow these steps:
+   More information on those methods can be found [here](#other-methods).
```

Comment:
```suggestion
   More information about these methods can be found [in section "other methods"](#other-methods).
```

Maybe rename section name?

Reply:
Renamed the section to "Optional Extractors" (like in the mlr3 manual) and created a own section for hotstarting.